### PR TITLE
New fields to the User entity.

### PR DESCRIPTION
**Pull request by salvoscala**
We should alter the user entity and have these fields:

- First Name
- Last Name
- Phone Number
- Postal Address
- Email

We should also think about an easy way to add billing profiles from the user edit page (reference_dialog?), a page where the user can manage his billing profiles and remove telephone from billing profile.

### CODE REVIEW

**Comment by salvoscala** [comment]
@cecrs, @acrollet could you please review this? The whole tickets of the project (https://github.com/Roomify/RfA/projects/2) have been done here. Sonia has qa'd it (https://roomify.atlassian.net/browse/RD-800) testing the functionality for new bookings, but we should be also sure that everything on existing sites it's going to work properly.

Many, many thanks!

**Review by cecrs — CHANGES_REQUESTED** [review]
When attempting to update an existing installation:
pulled issue-152
built the distro
ran composer install
ran dbup
when attempting to clear cache/revert features with drush, get the following error: Fatal error: Call to a member function override_option() on null in .../votingapi.views_default.inc on line 21

**Comment by acrollet** [comment]
Let's go ahead and get the tests passing :)

File: modules/roomify/roomify_accommodation_booking/roomify_accommodation_booking.module [hunk 1]
```diff
@@ -1612,6 +1676,107 @@ function roomify_accommodation_booking_form_bat_type_edit_form_alter(&$form, &$f
...
+ *
+ * FORM_ID = user_register_form.
+ */
+function roomify_accommodation_booking_form_user_register_form_alter(&$form, &$form_state) {
+  $form['user_first_name']['#weight'] = -100;
+  $form['user_last_name']['#weight'] = -99;
+  $form['user_telephone']['#weight'] = -98;
+  if (strpos($form['#action'], 'render=references-dialog') !== FALSE) {
+    $form['account']['name']['#required'] = FALSE;
+    $form['account']['name']['#description'] = t('Using this (or email) the user will be able to access to the site.') . ' ' . $form['account']['name']['#description'];
```
Comment (acrollet) [inline modules/roomify/roomify_accommodation_booking/roomify_accommodation_booking.module]: let's change to "Using this username (or their email) the user will be able to access the site."

File: modules/roomify/roomify_system/roomify_system.module [hunk 2]
```diff
@@ -969,6 +970,9 @@ function roomify_system_roomify_rights() {
       'view any commerce_customer_profile entity',
       'create commerce_customer_profile entities',
       'view own property_owner profile',
+      // Authenticated users.
+      'edit users with role 2',
```
Comment (acrollet) [inline modules/roomify/roomify_system/roomify_system.module]: where does this permission come from? is there any way to use a machine name?

File: modules/roomify/roomify_system/roomify_system.module [hunk 3]
```diff
@@ -1121,6 +1126,10 @@ function roomify_system_roomify_rights() {
     'administrator' => $permissions,
   );
 
+  if ($guest_role = user_role_load_by_name('guest')) {
+    $rights['roomify_system']['property owner'][] = 'edit users with role ' . $guest_role->rid;
```
Comment (acrollet) [inline modules/roomify/roomify_system/roomify_system.module]: ah-ha! ;)

**Review by acrollet — APPROVED** [review]
@salvoscala CR passes with fixes for these small things :)

**Pull request merged**